avogadro/core/cube.cpp

```diff
@@ -7,21 +7,17 @@
 
 #include "molecule.h"
 #include "mutex.h"
+#include <memory>
 
 namespace Avogadro::Core {
 
-Cube::Cube()
-  : m_data(0), m_min(0.0, 0.0, 0.0), m_max(0.0, 0.0, 0.0),
-    m_spacing(0.0, 0.0, 0.0), m_points(0, 0, 0), m_minValue(0.0),
-    m_maxValue(0.0), m_lock(new Mutex)
-{
-}
+    Cube::Cube()
+            : m_data(0), m_min(0.0, 0.0, 0.0), m_max(0.0, 0.0, 0.0),
+              m_spacing(0.0, 0.0, 0.0), m_points(0, 0, 0), m_minValue(0.0),
+              m_maxValue(0.0),
+              m_lock(std::make_unique<Mutex>()) // Initializes m_lock with a new Mutex instance
```

Comment:
The basic idea is good, but if you're going to use a `unique_ptr` you'll need to change `m_lock` in the header as well as the relevant methods and calling code.

Right now it doesn't compile, for example.